samples: doc: Added README for RTIO sample (#82373)
zephyrproject-rtos/zephyr: samples/subsys/rtio/sensor_batch_processing/README.rst

.. zephyr:code-sample:: sensor_batch_processing
   :name: Sensor batch processing
   :relevant-api: rtio

>    Implements sensor device using RTIO for asynchronuous data processing.

Reviewer: a typo + changing wording to follow same style as other samples (https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/templates/sample.tmpl#L5-L6)

```suggestion
   Implement a sensor device using RTIO for asynchronous data processing.
```

Author: Fixed

Overview
********

> This sample application demonstrates the use of the :ref:`rtio` library
> which is a framework for doing asynchronuous operation chains.

Reviewer: same typo :) + suggested wording improvement
```suggestion
This sample application demonstrates the use of the :ref:`rtio` framework for
doing asynchronuous operation chains.
```

Author: Fixed

Application uses :ref:`rtio` with mempool API to fetch data from virtual sensor
and displays it on the console.

Requirements
************

* A board with flash support or native_sim target

Building and Running
********************

This sample can be found under :zephyr_file:`samples/subsys/rtio` in the Zephyr tree.

The sample can be built for most platforms, the following commands build the
application for the native_sim target.

.. zephyr-app-commands::
   :zephyr-app: samples/subsys/rtio
   :board: native_sim
   :goals: build run
   :compact:

> When running the output on the console shows the operations of
> submitting for, consuming and processing the sensor data.

Reviewer: ```suggestion
When running, the output on the console shows the operations of
submitting for, consuming, and processing the sensor data.
```

Author: Fixed
